### DISCUSSION TITLE
Strange behavior of indents in labels (using tokens)

### DISCUSSION THREAD

**Discussion by nymver** [polybar/polybar#3069] [Q&A]
So I have a module that displays the artist and title of current playing song. I have specified tokens (minimum and maximum width) (label = %output:-45:45%). If there are no Cyrillic letters in the text, everything is fine, but if there are, for some reason the width of the label is reduced.

Module:
```
[module/running_string]
type = custom/script
exec = $HOME/.config/polybar/scripts/running_string
tail = true
format = <label>
label = %output:-45:45%
```

![Screenshot_2024-01-04_02 42 18](https://github.com/polybar/polybar/assets/146001324/389d834b-c44b-482b-a739-c1b2fdec1c4c)
![Screenshot_2024-01-04_02 42 24](https://github.com/polybar/polybar/assets/146001324/8b326d91-9391-4013-9306-0a958c0665c9)

**Comment by patrick96** [comment]
What fonts are you using?

**Reply by nymver** [reply]
CodeNewRoman Nerd

**Reply by perdakovich** [reply]
It happens with any font, I think.
I have Iosevka NFP and it's the same.

**Reply by patrick96** [reply]
Ah, I see what's happening here. The min and max width calculations kind of fall apart if you use non-ASCII characters.

For max-width, it properly counts the number of unicode codepoints while for the min-width it just counts the bytes. Cyrillic letters probably take up 2 bytes each, so it thinks the text is twice as long as it actually is.
This is something we should fix and just use the unicode length in all places

**Reply by patrick96** [reply]
Tracked in #3074

**Comment by nklloyd** [comment]
Hi, I would love to help with this issue.

**Reply by patrick96** [reply]
@nklloyd That's great! The relevant code is in `label.cpp` in `label::replace_token`. There the part dealing with the minimum width should deal with the number of utf8 codepoints (`string_util:.char_len`) instead of just the string length.

Let me know if you need help with anything.

**Reply by nklloyd** [reply]
Sounds good, thank you!

**Reply by nklloyd** [reply]
I believe I have corrected the issue. This is my first time contributing to a project like this, so if you could give me some pointers on how best to test out the fix that would be great. I did read through the installation documentation and the contributing guidelines. Should I install the latest PolyBar on my computer and then replace the label.cpp file with my updated local file? Or is it possible for me to test it in an IDE such as CLION? Thanks!
